src/core/WorkflowContext.ts

```diff
@@ -272,9 +281,50 @@ export class WorkFlowContext {
 		return escodegen.generate(ast, opts);
 	}
 
+	public replaceTSPathAliases(requireStatement: string): { requireStatement: string; replaced: boolean } {
+		let replaced = false;
+
+		const { compilerOptions } = this.tsConfig.getConfig();
+		const normalizedRequireStmt = requireStatement.replace(/\\/g, "/");
+		const { resolvedModule } = ts.resolveModuleName(
+			normalizedRequireStmt,
+			"",
+			compilerOptions,
+			ts.sys,
+			this.tsModuleResolutionCache,
+		);
+
+		if (!resolvedModule) return { requireStatement, replaced };
+
+		if (resolvedModule.packageId) {
+			requireStatement = resolvedModule.packageId.name;
+			replaced = true;
+		} else if (resolvedModule.resolvedFileName) {
+			requireStatement = path.relative(this.homeDir, resolvedModule.resolvedFileName);
+
+			if (resolvedModule.extension) {
+				requireStatement = requireStatement.substr(0, requireStatement.length - resolvedModule.extension.length);
+			}
+
+			requireStatement = `~/${requireStatement}`;
+
+			replaced = true;
+		}
+
+		return { requireStatement, replaced };
+	}
+
 	public replaceAliases(requireStatement: string): { requireStatement: string; replaced: boolean } {
 		const aliasCollection = this.aliasCollection;
 		let replaced = false;
+
+		if (this.automaticTSPathAlias && this.tsPathsRegExp && this.tsPathsRegExp.test(requireStatement)) {
```

Comment:
Shall we also check also that we are in a typescript file?  We don't want to do that on node_modules

Reply:
Added that :)